### ISSUE TITLE
Data Catalog Entry expand allowed type

### ISSUE THREAD

**Issue by DaveyDevOps** [hashicorp/terraform-provider-google#17653]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team) to expedite investigation and resolution of this issue.

### Description

[type](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/data_catalog_entry#type) - (Optional) The type of the entry. Only used for Entries with types in the EntryType enum. Currently, only FILESET enum value is allowed. All other entries created through Data Catalog must use userSpecifiedType.

I believe this enum can be expanded to include 'DATA_STREAM' and 'CLUSTER'. 

### New or Affected Resource(s)

* google_data_catalog_entry

### Potential Terraform Configuration

```tf 
resource "google_data_catalog_entry" "pubsub" {
  entry_group         = "projects/${myproject}/locations/global/entryGroups/@pubsub"
  entry_id            = base64encode(pubsub.id)
  type = "DATA_STREAM"
  linked_resource = pubsub.id
  schema          = file.json
```

### References

You can create entries only with 'FILESET', 'CLUSTER', 'DATA_STREAM', or custom types. Data Catalog automatically creates entries with other types during metadata ingestion from integrated systems.

https://cloud.google.com/data-catalog/docs/reference/rest/v1/projects.locations.entryGroups.entries/create

b/331229335

**Comment by DaveyDevOps** [comment]
I'm interested in working on this issue. Thanks

**Comment by ScottSuarez** [comment]
Hi @DaveyDevOps, I'll assign this ticket to you for now

**Comment by melinath** [comment]
Note from triage: We may want to just remove validation and convert this to a string field, since there will likely be more types added in the future.

**Comment by DaveyDevOps** [comment]
Although the resource supports many values in the type field the create operation is limited to a specific subset where
"metadata ingestion from integrated systems" is not yet possible. 

My main interest is in the DATA_STREAM type because Dataflow SQL requires schema entries in the Data Catalog when working with PubSub sources.

**Comment by DaveyDevOps** [comment]
Findings

[Beta documentation](https://cloud.google.com/data-catalog/docs/reference/rest/v1beta1/projects.locations.entryGroups.entries#EntryType) still states _"Currently, only FILESET enum value is allowed. All other entries created through Data Catalog must use userSpecifiedType."_

Creating entries with `type = "DATA_STREAM"` or `type = "CLUSTER"` was supported but these result in Kafka topic or Kafka cluster respectively. 

![Screenshot from 2024-04-27 08-01-41](https://github.com/hashicorp/terraform-provider-google/assets/34562546/847a4592-65ef-45df-aa75-f0ae3fc0106e)
![Screenshot from 2024-04-27 08-00-29](https://github.com/hashicorp/terraform-provider-google/assets/34562546/894ef5f7-f5f2-4d6c-90cd-36e914ad94b3)

I did not locate a method to set the values like Bootstrap servers, Cluster Entry, Topic. 

I found limited support for management of these entries in the Cloud Console and had to rely on the API explorer. 

Therefore I suspect support for these types was added for the development and integration of [Managed Apache Kafka](https://cloud.google.com/products/apache-kafka-for-bigquery) rather than direct use. 

Suggest close issue with no code changes. Thanks
